src/librustc_passes/rvalue_promotion.rs

```diff
@@ -169,12 +168,7 @@ impl<'a, 'gcx> CheckCrateVisitor<'a, 'gcx> {
     }
 }
 
-impl<'a, 'tcx> Visitor<'tcx> for CheckCrateVisitor<'a, 'tcx> {
```

Comment:
I should note that this part of the change *makes no sense* without #52318.
Please motivate such changes better in the future, using the PR description.